### DISCUSSION TITLE
A few thoughts

### DISCUSSION THREAD

**Discussion by richelo** [alexjustesen/speedtest-tracker#254] [Ideas]
I first want to say thank you so much for picking this one up since the "other one" was abandoned.

So, a few thoughts/ideas. Apologies if there are duplicates from other requests ...

- Dark mode selection. This one is weird ... On my laptop, it is dark mode, while on my desktop, it is light mode. Both are running macOS in dark mode permanently. So, it would be great to set it in settings.
- Min, Max, and average for all data like the other one.
- Export data
- Import data
- Reset data
- Either direct Home Assistant integration, or at the very least MQTT

That is all I have for now, but I am sure there will be more ;-)

Thanks again for your hard work on this!

Cheers

Rich

**Comment by alexjustesen** [comment]
Thanks for the feedback Rich, a couple of notes below for your bullet points.

- > Dark mode selection. This one is weird ... On my laptop, it is dark mode, while on my desktop, it is light mode. Both are running macOS in dark mode permanently. So, it would be great to set it in settings.
    - I don't have a MAC to test this but you can toggle between light/dark in the top right profile dropdown. If it continues to be an issue please open an Issue and I can engage the devs of Filament as well.
- > Min, Max, and average for all data like the other one.
    - This is def on the roadmap but not as high of a priority as the next couple of items.
- > Export / import / reset data
    - This is my next priority now that "pretty charts" have been released. This includes being able to import data from the "other" Speedtest Tracker.
- > Either direct Home Assistant integration, or at the very least MQTT
    - Can you give me an example of how this would be useful? It won't be included in `v1.0.0` but if this is an integration that makes sense I can look into it for a version after `1.0.0`.

Thanks!

**Reply by richelo** [reply]
Hey Alex,

I don't see a dark mode option at all on my desktop:
![image](https://user-images.githubusercontent.com/8215529/209100305-0c3ee0aa-0b36-4091-b316-3fba1977eca0.png)

As an FYI, I am running the latest version of MacOS and Chrome.

On the same machine, using the Orion browser, it DOES work:
![image](https://user-images.githubusercontent.com/8215529/209101870-788b88ae-b5ee-4229-a766-11ff0b02b833.png)

So this is something with the latest Chrome on this machine. On the laptop, same MacOS, and same Chrome, it works. UGH, the JOYS.

For the HA integration or MQTT ... With this we can run automations.

For instance, download speed is less that X, and time is between y and z of day, flash light in house and send email.

The options are endless for automation.

Direct HA integration would be great, but simple MQTT would be more than sufficient.

Thanks again for your great work!

**Reply by alexjustesen** [reply]
I haven't really played with HA too much (it's my upcoming winter project) but that sounds like a cool idea. Definitely something I can look into after v1.

**Reply by alexjustesen** [reply]
Also on the dark mode thing, I shared this with the Filament Discord so let's see if anyone else has experienced this.

**Comment by alexdelprete** [comment]
Regarding the dark mode: I think he's referring to the mode autodetection, it seems that in his case on one of the two systems dark mode of the OS setting is not being detected.

I agree that maybe an option to force light/dark mode in the settings would be appropriate. Personally, I usually force dark mode in the settings, even if the autodetection feature is working fine.

**Reply by alexjustesen** [reply]
So even if darkmode is/isn't detected he should still have the option to toggle it.

**Reply by alexdelprete** [reply]
yes, even if most apps have Auto as default based on OS setting, I always force it to Dark anyway. And in his case, since autodetection doesn't work on his specific environment, would solve the issue.

**Comment by darianf01** [comment]
Jumping in as well with an upvote from my side for the HA integration. Would love to see some integration with home assistant where API's of speedtest tracker are exposed so that HA can fetch results of past execution and display in HA. From HA, you can do lot's of automations based on results...
Also to be able to trigger API for queueing the test.
Maybe this is already possible? but not sure how can the speedtest tracker APIs be accessed from HA...

**Reply by BigFlubba** [reply]
Agreed! I use HA as my main dashboard and I have all of my internet gear monitored in HA. It would be nice to see cool graphs of my internet speeds in the dashboard directly.

**Reply by metaathron** [reply]
Hi. I have created my own HA integration. Feel free to test it if you're still interested and let me know if it suits your needs. https://github.com/metaathron/ha-speedtest-tracker

**Comment by phellarv** [comment]
I would very much like that Speedtest Tracker could post results to MQTT. Should be pretty easy to achieve.
